skfda/representation/grid.py

```diff
@@ -655,11 +738,19 @@ def concatenate(self, other):
 
         """
         # Checks
-        self.__check_same_dimensions(other)
+        for other in others:
+            self.__check_same_dimensions(other)
+
+        if as_coordinate:
+            if any([self.nsamples != other.nsamples for other in others]):
+                raise ValueError(f"All the FDataGrids must contain the same "
+                                 f"number of samples {self.nsamples} to "
+                                 f"concatenate as a new coordinate.")
+
+        data = [self.data_matrix] + [other.data_matrix for other in others]
+        axis = 0 if as_coordinate is False else -1
 
-        return self.copy(data_matrix=numpy.concatenate((self.data_matrix,
-                                                        other.data_matrix),
-                                                       axis=0))
+        return self.copy(data_matrix=numpy.concatenate(data, axis=axis))
```

Comment:
The labels must also be changed accordingly.

Reply:
Added in fa02b84 and 970e731:
* Logic of axes_labels moved to `FData` with a property with setter
* Methods to get the labels of the coordinates indexed
* Test of indexation